submissions/AnasImloul/main.cpp

```diff
@@ -61,107 +59,106 @@ std::string fruits_and_vegetables[94] = {
         map[city].reserve(512);
         for (const auto& product : fruits_and_vegetables)
-            map[city][product] = INT_MAX;
+            map[city][product] = INT64_MAX;
     }
 
-    std::unordered_map<std::string, double> index_totals;
-    for (const auto& city : morrocan_cities) {
-        index_totals[city] = 0;
+    // map to store the total price of each city (sum of all products)
+    unordered_map<string, int64_t> total_per_city;
+    total_per_city.reserve(613);
+    for (const auto& city : moroccan_cities) {
+        total_per_city[city] = 0;
     }
 
-    std::string city, product;
-    char c;
-
-    int count = 0;
+    ofstream output("output.txt");
 
-    while (!std::cin.eof()) {
+    string city, product;
+    char c;
+    while (!cin.eof()) {
         city.clear(), product.clear();
 
-        while (std::cin.get(c) && c != ',') city.push_back(c);
+        while (cin.get(c) && c != ',') city.push_back(c);
 
-        while (std::cin.get(c) && c != ',') product.push_back(c);
+        while (cin.get(c) && c != ',') product.push_back(c);
 
         int price = 0;
-        while (std::cin.get(c) && c != '\n') {
+        while (cin.get(c) && c != '\n') {
             if (c == '.') break;
             else price = price * 10 + (c - '0');
         }
 
         int decimal = 0;
         if (c == '.') {
-            while (std::cin.get(c) && c != '\n') {
+            while (cin.get(c) && c != '\n') {
                 price = price * 10 + (c - '0');
                 decimal++;
             }
         }
 
         if (price == 0) break;
 
-        double price_double = price;
-        if (decimal == 1) price_double /= 10;
-        else if (decimal == 2) price_double /= 100;
+        if (decimal == 0) price *= 100;
+        else if (decimal == 1) price *= 10;
 
-        double& current_price = map[city][product];
-        if (price_double < current_price)
-            current_price = price_double;
-
-        index_totals[city] += price_double;
-
-        count++;
+        if (price < 1000) {
+            int64_t &current_price = map[city][product];
+            if (current_price == 0 || price < current_price) current_price = price;
+        }
+        total_per_city[city] += price;
     }
 
-    std::ofstream output("output.txt");
-
     // get city with the lowest price
-    std::string cheapest_city;
-    double cheapest_price = std::numeric_limits<double>::max();
-
-    for (const auto& [city, total] : index_totals) {
-        if (total < cheapest_price) {
-            cheapest_price = total;
-            cheapest_city = city;
-        }
+    pair<int64_t, string> cheapest = {INT64_MAX, ""};
+    for (const auto& [city, total] : total_per_city) {
+        if (total == 0) continue;
+        cheapest = min(cheapest, {total, city});
     }
 
-    output << cheapest_city << " " << std::fixed << std::setprecision(2) << cheapest_price << std::endl;
+    // print the city with the lowest price
+    output << cheapest.second
```

Comment:
Are you sure about this ?

Reply:
Thanks for the remark, I had to do it at one point of the program, but now it's not necessary. I am gonna fix it in a quick commit

Reply:
have you tested it ? Please do not use this as your playground 😅 we expect you to submit a fully costumised and optimised code

Reply:
Yes don't worry I did test it :) I do realize that the judge server is not a made for testing for obvious reasons (test_count * 900sec per submission). The output formatting I did doesn't change the output of the program. I should have just said it's correct even though it's ugly 😅